src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/transform/resthandler/RestStopTransformActionIT.kt

```diff
@@ -0,0 +1,208 @@
+    }
+
+    @Throws(Exception::class)
+    fun `test stopping a finished transform`() {
+        // Create a transform that finishes
+        val transform = createTransform(
+            randomTransform()
+                .copy(
+                    jobSchedule = IntervalSchedule(Instant.now(), 1, ChronoUnit.MINUTES),
+                    enabled = true,
+                    enabledAt = Instant.now(),
+                    metadataId = null
+                )
+        )
+        createTransformSourceIndex(transform)
+        updateTransformStartTime(transform)
+
+        // Assert it finished
+        waitFor {
+            val updatedTransform = getTransform(transform.id)
+            val metadata = getTransformMetadata(updatedTransform.metadataId!!)
+            assertEquals("Transform never finished", TransformMetadata.Status.FINISHED, metadata.status)
+            // Waiting for job to be disabled here to avoid version conflict exceptions later on
+            assertFalse("Job was not disabled", updatedTransform.enabled)
+        }
+
+        // Try to stop a finished transform
+        val response = client().makeRequest("POST", "$TRANSFORM_BASE_URI/${transform.id}/_stop")
+        assertEquals("Stop transform failed", RestStatus.OK, response.restStatus())
+        val expectedResponse = mapOf("acknowledged" to true)
+        assertEquals(expectedResponse, response.asMap())
+
+        // Assert it is still in finished status
+        waitFor {
+            val updatedTransform = getTransform(transform.id)
+            val metadata = getTransformMetadata(updatedTransform.metadataId!!)
+            assertEquals("Transform should have stayed finished", TransformMetadata.Status.FINISHED, metadata.status)
+        }
+    }
+
+    @Throws(Exception::class)
+    fun `test stopping a failed transform`() {
+        // Create a transform that will fail because no source index
+        val transform = randomTransform().copy(
+            id = "test_stopping_a_failed_transform",
+            jobSchedule = IntervalSchedule(Instant.now(), 1, ChronoUnit.MINUTES),
+            enabled = true,
+            enabledAt = Instant.now(),
+            metadataId = null
+        ).let { createTransform(it, it.id) }
+        updateTransformStartTime(transform)
+
+        // Assert it's in failed
+        waitFor {
+            val updatedTransform = getTransform(transform.id)
+            val metadata = getTransformMetadata(updatedTransform.metadataId!!)
+            assertEquals("Transform never failed", TransformMetadata.Status.FAILED, metadata.status)
+        }
+
+        // Stop transform
+        val response = client().makeRequest("POST", "$TRANSFORM_BASE_URI/${transform.id}/_stop")
+        assertEquals("Stop transform failed", RestStatus.OK, response.restStatus())
+        val expectedResponse = mapOf("acknowledged" to true)
+        assertEquals(expectedResponse, response.asMap())
```

Comment:
Can you move this inside waitFor{} block, I am seeing transient failures for test too when I run the test in a loop